pkg/qbittorrent/client.go

```diff
@@ -242,3 +263,50 @@ func (c *Client) setCookies(cookies []*http.Cookie) {
 	cookieURL, _ := url.Parse(fmt.Sprintf("%v://%v:%v", c.settings.protocol, c.settings.Hostname, c.settings.Port))
 	c.http.Jar.SetCookies(cookieURL, cookies)
 }
+
+func buildUrl(settings Settings, endpoint string) string {
+	// parse url
+	u, _ := url.Parse(settings.Hostname)
+
+	// reset Opaque
+	u.Opaque = ""
+
+	// set scheme
+	scheme := "http"
+	if u.Scheme == "http" || u.Scheme == "https" {
+		if settings.TLS {
+			scheme = "https"
+		}
+		u.Scheme = scheme
+	} else {
+		if settings.TLS {
+			scheme = "https"
+		}
+		u.Scheme = scheme
+	}
+
+	// if host is empty lets use one from settings
+	if u.Host == "" {
+		u.Host = settings.Hostname
+	}
+
+	// reset Path
+	if u.Host == u.Path {
+		u.Path = ""
+	}
+
+	// handle ports
+	if settings.Port > 0 {
+		if settings.Port == 80 || settings.Port == 443 {
```

Comment:
Why?

Reply:
For the clients that still has a port we want to keep that. But in the case of 80 and 443 they're not needed and are stripped.

Again for compatibility, but maybe a bad idea?

Reply:
I haven't looked at the golang http client to see if there's a fast path or something like that for this. Adding branches like this not assigning a port just makes it more brittle long term. It's all minor, just sharing an opinion.